File: dab2c-executor-application/src/main/java/ru/sbrf/ufs/dab2c/core/integration/healthcheck/ApplicationHealthCheck.java
package ru.sbrf.ufs.dab2c.core.integration.healthcheck;

import org.springframework.stereotype.Component;
import ru.sbrf.ufs.platform.healthcheck.Health;
import ru.sbrf.ufs.platform.healthcheck.HealthCheck;

/**
 * Реализация мониторинга здоровья приложения
 */
@Component
public class ApplicationHealthCheck implements HealthCheck {

    @Override
    public Health check() {
        return Health.OK;
    }
}

File: executor-shared/health-check/src/main/kotlin/ru/sbrf/ufs/dab2c/core/executor/shared/healthcheck/ApplicationHealthCheck.kt
package ru.sbrf.ufs.dab2c.core.executor.shared.healthcheck

import org.slf4j.Logger
import ru.sbrf.ufs.platform.healthcheck.Health
import ru.sbrf.ufs.platform.healthcheck.HealthCheck
import ru.sbrf.ufs.platform.logger.LoggerFactory

/**
 * Реализация мониторинга здоровья приложения.
 */
class ApplicationHealthCheck : HealthCheck {
    override fun check(): Health {
        LOGGER.info("Health:check()")
        return Health.OK
    }

    internal companion object {
        private val LOGGER: Logger = LoggerFactory.getLogger(
            ApplicationHealthCheck::class.java
        )
    }
}
